tests/Microsoft.Identity.Test.Unit/WwwAuthenticateParametersTests.cs

```diff
@@ -56,6 +59,129 @@ public void CreateWwwAuthenticateResponse(string resource, string authorizationU
             Assert.IsNull(authParams.Error);
         }
 
+        [TestMethod]
+        [DataRow("AuthScheme", "")]
+        [DataRow("AuthScheme", "token68")]
+        [DataRow("AuthScheme", "auth-param1=token1, auth-param2=token2, auth-param3=token3")]
+        [DataRow("AuthScheme", "token68 auth-param1=token1, auth-param2=token2, auth-param3=token3")]
+        public void EnsureProperlyFormattedHeadersWithToken86DoNotFail(string scheme, string values)
+        {
+            // Arrange
+            HttpResponseMessage httpResponse = new HttpResponseMessage(HttpStatusCode.Unauthorized);
+            httpResponse.Headers.Add(WwwAuthenticateHeaderName, $"{scheme} {values}");
+
+            // Act
+            var authParams = WwwAuthenticateParameters.CreateFromAuthenticationHeaders(httpResponse.Headers, scheme);
+
+            // Assert
+            Assert.AreEqual(scheme, authParams.AuthScheme);
+
+            if (string.IsNullOrEmpty(values))
+            {
+                Assert.AreEqual(0, authParams.RawParameters.Count);
+            }
+            else if (values == "token68")
+            {
+                Assert.AreEqual("token68", authParams.RawParameters[scheme]);
+            }
+            else
+            {
+                Assert.AreEqual("token1", authParams.RawParameters["auth-param1"]);
+                Assert.AreEqual("token2", authParams.RawParameters["auth-param2"]);
+                Assert.AreEqual("token3", authParams.RawParameters["auth-param3"]);
+            }
+        }
+
+        [TestMethod]
+        [DataRow("AuthScheme", "realm=someRealm")]
+        [DataRow("AuthScheme", "realm=someRealm token68")]
+        [DataRow("AuthScheme", "realm=someRealm auth-param1=token1, auth-param2=token2, auth-param3=token3")]
+        [DataRow("AuthScheme", "realm=someRealm token68 auth-param1=token1, auth-param2=token2, auth-param3=token3")]

```

Comment:
I am not clear from the specs - what is the separator?
[This HTTP spec](https://httpwg.org/specs/rfc9110.html#rfc.section.11.6.1) says 
> each challenge can contain a comma-separated list of authentication parameters

And same with [different section](https://httpwg.org/specs/rfc9110.html#rfc.section.11.2) in the same spec
> The authentication scheme is followed by additional information necessary for achieving authentication via that scheme as either a comma-separated list of parameters or a single sequence of characters capable of holding base64-encoded information.

Mozilla has examples of both - [comma separated](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate#basic_authentication) and [with spaces](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate#syntax) after realm and token68.

This test case passes
`[DataRow("AuthScheme", "realm=someRealm, auth-param1=token1, auth-param2=token2, auth-param3=token3")]`
but this does not
`[DataRow("AuthScheme", "realm=someRealm, token68, auth-param1=token1, auth-param2=token2, auth-param3=token3")]`

Reply:
I think I found a bug in our `SplitWithQuotes` methods that seems to fail when spaces are included within the values. Ill see if I can resolve it.

Reply:
Ok, there was no bug there, I found the issue in the test code. 
In regards to the seperator, there doesn't seem to be a clear indicator of what it is in the spec. It seems like it can either be a space or a comma followed by a space. So, we are accounting for both of these scenarios.

Reply:
Thanks for verifying. Let's make sure we have test cases that cover these scenarios to protect from regressions.